libsrc/apple2/mli_file_info.s

```diff
@@ -0,0 +1,54 @@
+;
+; Colin Leroy-Mira, 2023 <colin@colino.net>
+;
+
+        .export         mli_file_info, mli_file_info_direct
+        .import         pushname, popname, return0
+        .import         popax
+        .import         _open,_close
+        .include        "zeropage.inc"
+        .include        "errno.inc"
+        .include        "fcntl.inc"
+        .include        "filedes.inc"
+        .include        "mli.inc"
+        .include        "stat.inc"
+
+        ; Calls ProDOS MLI GET_FILE_INFO on the ProDOS style
+        ; filename stored on top of stack
+        ; Returns with carry set on error, and sets errno
+mli_file_info_direct:
+        ; Set pushed name
+        lda     sp
+        ldx     sp+1
+        sta     mliparam + MLI::INFO::PATHNAME
+        stx     mliparam + MLI::INFO::PATHNAME+1
+
+        ; Get file information
+        lda     #GET_INFO_CALL
+        ldx     #GET_INFO_COUNT
+        jmp     callmli
+
+
+
+        ; Calls ProDOS MLI GET_FILE_INFO on the filename
+        ; stored as C string in AX at top of stack
+        ; Returns with carry set on error, and sets errno
+mli_file_info:
+        ; Get pathname
+        jsr     popax
+        jsr     pushname
+        bne     oserr
+
+        jsr     mli_file_info_direct
+        php                     ; Save return status
+
+        jsr     popname         ; Preserves A
+
+        plp
+        bcs     oserr
+        rts
+
+oserr:
+        jsr     ___mappederrno
+        sec
+        rts
```

Comment:
I don't see a technical reason why these two functions need to be in a single file. Therefore I ask you to separate them in order to have i.e. exec() callers not link the unused code of mli_file_info.

Reply:
Good idea!